lib/Server.js

```diff
@@ -3255,9 +3255,65 @@ class Server {
    * @param {string | string[]} watchPath
    * @param {WatchOptions} [watchOptions]
    */
-  watchFiles(watchPath, watchOptions) {
+  watchFiles(watchPath, watchOptions = {}) {
     const chokidar = require("chokidar");
-    const watcher = chokidar.watch(watchPath, watchOptions);
+
+    const watchPathArr = Array.isArray(watchPath) ? watchPath : [watchPath];
+
+    if (watchOptions.disableGlobbing !== true) {
+      const isGlob = require("is-glob");
+      const watchPathGlobs = watchPathArr.filter((p) => isGlob(p));
+
+      // No need to do all this work when no globs are used
+      if (watchPathGlobs.length > 0) {
+        const globParent = require("glob-parent");
+        const picomatch = require("picomatch");
+
+        watchPathGlobs.forEach((p) => {
+          watchPathArr[watchPathArr.indexOf(p)] = globParent(p);
+        });
+
+        const matcher = picomatch(watchPathGlobs, {
+          cwd: watchOptions.cwd,
+          dot: true,
+        });
+
+        const ignoreFunc = (/** @type {string} */ p) =>
+          !watchPathArr.includes(p) && !matcher(p);
+
+        const ignoredArr = Array.isArray(watchOptions.ignored)
+          ? watchOptions.ignored
+          : [];
+
+        // Nested ternaries are forbidden by eslint so we end up with this
+        if (watchOptions.ignored && !Array.isArray(watchOptions.ignored)) {
+          ignoredArr.push(watchOptions.ignored);
+        }
+
+        const ignoredGlobs = [];
+        for (let i = 0; i < ignoredArr.length; i++) {
+          const ignored = ignoredArr[i];
+          if (typeof ignored === "string" && isGlob(ignored)) {
+            ignoredGlobs.push(ignored);
+            ignoredArr.splice(i, 1);
+          }
+        }
+
+        if (ignoredGlobs.length > 0) {
+          const ignoreMatcher = picomatch(ignoredGlobs, {
+            dot: true,
+            cwd: watchOptions.cwd,
+          });
+          ignoredArr.push(ignoreMatcher);
+        }
+
+        ignoredArr.push(ignoreFunc);
+
+        watchOptions.ignored = ignoredArr;
```

Comment:
I would like to finish this soon and make a release

Reply:
Will do!

As long as chokidar is not handling the issue linked in the first post, this is blocked anyway. As alternative we have to filter out undefined options ourselves before passing them to chokidar. Wdyt?

Reply:
@alexander-akait I tried it your way. But there is one detail that might end up being performance sensitive.
I dont want to generate multiple matchers (one for each glob) if one is sufficient for all globs. Therefore I need to filter out the glob strings twice.

I tried to simplify the code further and think the code that I have now is easy enough to understand. I made it so that everything ends up in the ignoreArr.
Also fixed a bug while I was at it. 

We need tests for the ignore case though! I am not quite sure how to do the ignore case. I need to take a look.

Also I made the snapshots break because ignore is an array now when passied to chokidar. Need to fix that as well